doc/introduction/remote.rst

```diff
@@ -0,0 +1,511 @@
+
+We choose a random 4x4 interferometer
+
+>>> U = random_interferometer(4)
+>>> print(U)
+array([[-0.13879438-0.47517904j,-0.29303954-0.47264099j,-0.43951987+0.12977568j, -0.03496718-0.48418713j],
+[ 0.06065372-0.11292765j, 0.54733962+0.1215551j, -0.50721513+0.56195975j, -0.15923161+0.26606674j],
+[ 0.42212573-0.53182417j, -0.2642572 +0.50625182j, 0.19448705+0.28321781j,  0.30281396-0.05582391j],
+[ 0.43097587-0.30288974j, 0.07419772-0.21155126j, 0.28335618-0.13633175j, -0.75113453+0.09580304j]])
+
+Next we create the program
+
+.. code-block:: python3
+
+    prog = sf.Program(8)
+
+    with prog.context as q:
+        # Initial squeezed states
+        # Allowed values are r=1.0 or r=0.0
+        ops.S2gate(1.0) | (q[0], q[4])
+        ops.S2gate(1.0) | (q[1], q[5])
+        ops.S2gate(1.0) | (q[3], q[7])
+
+        # Interferometer on the signal modes (0-3)
+        ops.Interferometer(U) | (q[0], q[1], q[2], q[3])
+        ops.BSgate(0.543, 0.123) | (q[2], q[0])
+        ops.Rgate(0.453) | q[1]
+        ops.MZgate(0.65, -0.54) | (q[2], q[3])
+
+        # *Same* interferometer on the idler modes (4-7)
+        ops.Interferometer(U) | (q[4], q[5], q[6], q[7])
+        ops.BSgate(0.543, 0.123) | (q[6], q[4])
+        ops.Rgate(0.453) | q[5]
+        ops.MZgate(0.65, -0.54) | (q[6], q[7])
+
+        ops.MeasureFock() | q
+
+We create the engine. The engine is in charge of compiling and executing
+programs on the remote device.
+
+>>> eng = RemoteEngine("chip2")
+
+We run the engine by calling ``eng.run``, and pass it the program we
+want to run.
+
+>>> results = eng.run(prog, shots=20)
+Job e6ead866-04c9-4d48-ba28-680e8639fc41 is sent to server.
+>>> results.samples.T
+array([[0, 0, 1, 0, 1, 0, 1, 0],
+       [0, 0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 0, 2],
+       [0, 0, 0, 0, 0, 1, 0, 0],
+       [1, 0, 0, 0, 0, 0, 3, 0],
+       [3, 0, 0, 0, 2, 0, 1, 0],
+       [0, 1, 0, 0, 0, 1, 1, 0],
+       [0, 1, 0, 0, 1, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 1, 1],
+       [0, 0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 1, 0, 0],
+       [1, 0, 0, 0, 1, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 1, 0],
+       [0, 0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 0, 1],
+       [0, 0, 0, 0, 0, 0, 0, 1],
+       [1, 0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 1, 0, 0],
+       [0, 0, 1, 1, 0, 2, 1, 2],
+       [2, 0, 1, 0, 1, 0, 0, 0]])
+>>> np.mean(results.samples.T, axis=0)
+array([0.4 , 0.1 , 0.15, 0.05, 0.3 , 0.3 , 0.45, 0.35])
+
+
+We can convert the samples into counts using the following function:
+
+.. code-block:: python3
+
+     from collections import Counter
+
+     def count(samples):
+          bitstrings = [tuple(i) for i in samples]
+          return {k:v for k, v in Counter(bitstrings).items()}
+
+>>> samples = np.array([[0, 2],[1, 0],[0, 1],[0, 0],[0, 0],[2, 0],[0, 1],[0, 1]])
+>>> counts = count(samples)
+>>> print(counts)
+{(0, 2): 1, (1, 0): 1, (0, 1): 3, (0, 0): 2, (2, 0): 1}
+>>> counts[(0, 0)]
+2
```

Comment:
We have an issue to add this 😆 #157 The reason it is added here is because people were leaving tickets asking how to do this during the alpha period, so I thought best to add it to a central location.